Source/Lib/Codec/EbPictureDecisionProcess.c

```diff
@@ -1009,20 +1004,16 @@ void* PictureDecisionKernel(void *inputPtr)
             }
             // If an #IntraPeriodLength has passed since the last Intra, then introduce a CRA or IDR based on Intra Refresh type
             else if(sequenceControlSetPtr->intraPeriodLength != -1) {
-                pictureControlSetPtr->craFlag = 
-                    (sequenceControlSetPtr->intraRefreshType != CRA_REFRESH) ?
-                        pictureControlSetPtr->craFlag :
-                        (encodeContextPtr->intraPeriodPosition == (EB_U32) sequenceControlSetPtr->intraPeriodLength) ? 
-                            EB_TRUE : 
-                            pictureControlSetPtr->craFlag;
-
-                pictureControlSetPtr->idrFlag = 
-                    (sequenceControlSetPtr->intraRefreshType != IDR_REFRESH) ? 
-                        pictureControlSetPtr->idrFlag :
-                        (encodeContextPtr->intraPeriodPosition == (EB_U32) sequenceControlSetPtr->intraPeriodLength) ? 
-                            EB_TRUE : 
-                            pictureControlSetPtr->idrFlag;
-                
+                if ((encodeContextPtr->intraPeriodPosition == (EB_U32) sequenceControlSetPtr->intraPeriodLength) ||
+                    (pictureControlSetPtr->sceneChangeFlag == EB_TRUE)) {
+                    if (sequenceControlSetPtr->intraRefreshType == CRA_REFRESH) {
+                        pictureControlSetPtr->craFlag = EB_TRUE;
+                        pictureControlSetPtr->idrFlag = EB_FALSE;
+                    } else {
+                        pictureControlSetPtr->craFlag = EB_FALSE;
+                        pictureControlSetPtr->idrFlag = EB_TRUE;
+                    }
+                }
```

Comment:
Refactor for better readability?
Thanks.

Reply:
It follows the current code style.